src/main/scala/org/constellation/snapshotstreaming/SnapshotProcessor.scala

```diff
@@ -202,111 +217,142 @@ object SnapshotProcessor {
-                          Map.empty,
-                          LocalDateTime.now()
-                        )
-                      )
-                    )
-                    .map(_.toList)
-                case None =>
-                  new Throwable(
-                    s"Neither last processed snapshot nor initial snapshot were found on disk!"
-                  )
-                    .raiseError[F, List[GlobalSnapshotWithState]]
+            }
+          }
+          .evalTap { snapshots =>
+            logger.info(s"Producer: Processing ${snapshots.size} snapshots") >>
+              snapshots.traverse { case GlobalSnapshotWithState(snapshot, _, _, _, _) =>
+                logger.info(s"Producer: Offering snapshot to queue: ${getSnapshotReference(snapshot).show}")
               }
           }
-        }
-        .evalTap { snapshots =>
-          snapshots.traverse { case GlobalSnapshotWithState(snapshot, _, _, _, _) =>
-            logger.info(s"Pulled following global snapshot: ${getSnapshotReference(snapshot).show}")
+          .flatMap(Stream.emits)
+          .evalMap { snapshot =>
+            val hasher = HasherSelector[F].getForOrdinal(snapshot.snapshot.ordinal)
+            logger.info(s"Producer: Processing snapshot ${getSnapshotReference(snapshot.snapshot)}") >>
+              process(snapshot, hasher).handleErrorWith { e =>
+                logger.error(e)(
+                  s"Producer: Error processing snapshot ${getSnapshotReference(snapshot.snapshot)}, skipping"
+                ) >>
+                  ().pure[F]
+              }
+                .as(snapshot)
           }
-        }
-        .evalMap {
-          _.tailRecM {
-            case (state @ GlobalSnapshotWithState(snapshot, _, _, _, _)) :: nextSnapshots
-                if configuration.node.terminalSnapshotOrdinal.forall(snapshot.ordinal <= _) =>
-              val hasher = HasherSelector[F].getForOrdinal(snapshot.ordinal)
-              process(state, hasher) >> {
-                val snapshotWithState = SnapshotWithState(state.snapshot, state.snapshotInfo)
-                val snapshotOrdinal = state.snapshot.ordinal.value.value
+          .evalTap { snapshot =>
+            queue.offer(snapshot).flatMap { _ =>
+              logger.info(s"Producer: Added snapshot to queue (offered ${getSnapshotReference(snapshot.snapshot)})")
+            }
+          }
+          .drain
+
+        // Consumer stream - stores snapshots
+        consumer = Stream
+          .fromQueueUnterminated(queue)
+          .evalTap { snapshot =>
+            queue.size.flatMap { size =>
+              logger.info(
+                s"Consumer: Starting to process snapshot ${getSnapshotReference(snapshot.snapshot)}. Queue size: $size"
+              )
+            }
+          }
+          .evalMap { snapshot =>
+            val hasher = HasherSelector[F].getForOrdinal(snapshot.snapshot.ordinal)
+            store(snapshot, hasher)
```

Comment:
Is this store lazily evaluated ? Need to load this in IDE, but this call isn't pipelined with `>>`

Reply:
Yeah it's an F return, so need >>